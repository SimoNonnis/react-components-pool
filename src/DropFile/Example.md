DropFile with default props

```jsx
<DropFile />
```

DropFile with acceptText prop

```jsx
<DropFile acceptText="Only PDF allowed" />
```

DropFile with acceptText prop and errorMessage

```jsx
<DropFile acceptText="Only PDF allowed" errorMessage="Images are not allowed" />
```
