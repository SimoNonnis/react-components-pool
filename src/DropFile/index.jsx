import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

import {
  StyledDropzone,
  Container,
  StyledUploadIcon,
  Label,
  PickFileButton,
  ContainerText,
  Message,
  AttentionIcon,
} from './style';

function DropFile({
  acceptText,
  accept,
  onDrop,
  onDropAccepted,
  onDropRejected,
  errorMessage,
  dropAreaLabel,
  buttonLabel,
  multiple,
  className,
  ...props
}) {
  const dropAreaLabelComp = typeof dropAreaLabel === 'function' ? dropAreaLabel() : dropAreaLabel;
  const buttonLabelComp = typeof buttonLabel === 'function' ? buttonLabel() : buttonLabel;

  return (
    <div>
      <StyledDropzone
        activeStyle={{ borderColor: '#0099cf' }}
        accept={accept}
        onDrop={onDrop}
        onDropAccepted={onDropAccepted}
        onDropRejected={onDropRejected}
        multiple={multiple}
        className={className}
        {...props}
      >
        <Container>
          <StyledUploadIcon />
          <Label>{dropAreaLabelComp}</Label>
          <PickFileButton styleType="secondary" onClick={noop}>
            {buttonLabelComp}
          </PickFileButton>
          {!!errorMessage || !!acceptText ? (
            <ContainerText>
              {!!acceptText && <Message>{acceptText}</Message>}
              {!!errorMessage && (
                <Message error>
                  <AttentionIcon width={15} height={15} />
                  {errorMessage}
                </Message>
              )}
            </ContainerText>
          ) : null}
        </Container>
      </StyledDropzone>
    </div>
  );
}

DropFile.propTypes = {
  acceptText: PropTypes.string,
  accept: PropTypes.string,
  onDrop: PropTypes.func,
  onDropAccepted: PropTypes.func,
  onDropRejected: PropTypes.func,
  errorMessage: PropTypes.string,
  dropAreaLabel: PropTypes.oneOfType([PropTypes.func, PropTypes.node, PropTypes.string]),
  buttonLabel: PropTypes.oneOfType([PropTypes.func, PropTypes.node, PropTypes.string]),
  multiple: PropTypes.bool,
  className: PropTypes.string,
};

DropFile.defaultProps = {
  acceptText: null,
  accept: 'image/png, image/jpeg, application/pdf',
  onDrop: noop,
  onDropAccepted: noop,
  onDropRejected: noop,
  errorMessage: null,
  dropAreaLabel: 'Drop here to upload or',
  buttonLabel: 'Browse files',
  multiple: false,
  className: null,
};

export default DropFile;
