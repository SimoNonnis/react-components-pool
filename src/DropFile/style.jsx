import styled from 'styled-components';

import Dropzone from 'react-dropzone';

import UploadIcon from 'components/Icons/UploadIcon';
import Attention from 'components/Icons/Attention';
import P from 'components/Typography/P';
import Button from 'components/Buttons/Button';

export const StyledDropzone = styled(Dropzone)`
  border: 1px dashed ${({ theme }) => theme.color.mediumGrey};
  text-align: center;
  padding: ${({ theme }) => theme.spacing.lg} ${({ theme }) => theme.spacing.sm};
  cursor: pointer;
  background-color: ${({ theme }) => theme.color.white};
  width: 100%;
  box-sizing: border-box;
`;

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.color.veryDarkBlue};
  font-size: 16px;
  font-weight: 400;
`;

export const StyledUploadIcon = styled(UploadIcon)`
  color: ${({ theme }) => theme.color.lightGrey4};
`;

export const Label = styled(P)`
  margin-top: ${({ theme }) => theme.spacing.md};
`;

export const PickFileButton = styled(Button)`
  margin-top: ${({ theme }) => theme.spacing.sm};
`;

export const ContainerText = styled.div`
  padding: ${({ theme }) => theme.spacing.md};
  padding-bottom: 0;
  text-align: center;
`;

export const Message = styled(P)`
  color: ${({ error, theme }) => (error ? theme.color.red : theme.color.mediumGrey)};
  font-size: ${({ theme }) => theme.base.fontSize};
  font-style: ${({ error }) => (error ? 'normal' : 'italic')};
  margin-bottom: 0;
  margin-top: ${({ error, theme }) => (error ? theme.spacing.sm : '0')};
  ${({ error }) =>
    error &&
    `display: flex;
     align-items: center;
     justify-content: center;
  `};
`;

export const AttentionIcon = styled(Attention)`
  padding-right: ${({ theme }) => theme.spacing.xs};
  color: ${({ theme }) => theme.color.red};
`;
