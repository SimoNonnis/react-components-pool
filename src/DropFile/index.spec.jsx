/* eslint-env jest */
import React from 'react';

import DropFile from './';

describe('Test <DropFile />', () => {
  const mockFunction = jest.fn();
  it('should render DropFile correctly with default props', () => {
    const wrapper = shallow(<DropFile />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with acceptText prop', () => {
    const wrapper = shallow(<DropFile acceptText="Only images are allowed" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with accept prop', () => {
    const wrapper = shallow(<DropFile accept="application/pdf" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with onDrop prop', () => {
    const wrapper = shallow(<DropFile onDrop={mockFunction} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with onDropAccepted prop', () => {
    const wrapper = shallow(<DropFile onDropAccepted={mockFunction} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with onDropRejected prop', () => {
    const wrapper = shallow(<DropFile onDropRejected={mockFunction} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with errorMessage prop', () => {
    const wrapper = shallow(<DropFile errorMessage="Pdf are not allowed" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with dropAreaLabel prop', () => {
    const wrapper = shallow(<DropFile dropAreaLabel="Test Label" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with dropAreaLabel prop as a function', () => {
    const wrapper = shallow(
      <DropFile dropAreaLabel={() => <span>dropAreaLabel component</span>} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with buttonLabel prop', () => {
    const wrapper = shallow(<DropFile buttonLabel="Test buttonLabel" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with buttonLabel prop', () => {
    const wrapper = shallow(<DropFile buttonLabel={() => <span>buttonLabel component</span>} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with multiple prop', () => {
    const wrapper = shallow(<DropFile multiple />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render DropFile correctly with className prop', () => {
    const wrapper = shallow(<DropFile className="testClass" />);
    expect(wrapper).toMatchSnapshot();
  });
});
