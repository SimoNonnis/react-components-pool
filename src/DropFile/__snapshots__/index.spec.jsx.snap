// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <DropFile /> should render DropFile correctly with accept prop 1`] = `
<div>
  <Styled(t)
    accept="application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with acceptText prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
      <styled.div>
        <Styled(styled.p)>
          Only images are allowed
        </Styled(styled.p)>
      </styled.div>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with buttonLabel prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Test buttonLabel
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with buttonLabel prop 2`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        <span>
          buttonLabel component
        </span>
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with className prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className="testClass"
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with default props 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with dropAreaLabel prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Test Label
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with dropAreaLabel prop as a function 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        <span>
          dropAreaLabel component
        </span>
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with errorMessage prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
      <styled.div>
        <Styled(styled.p)
          error={true}
        >
          <Styled(Attention)
            height={15}
            width={15}
          />
          Pdf are not allowed
        </Styled(styled.p)>
      </styled.div>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with multiple prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={true}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with onDrop prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[MockFunction]}
    onDropAccepted={[Function]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with onDropAccepted prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[MockFunction]}
    onDropRejected={[Function]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;

exports[`Test <DropFile /> should render DropFile correctly with onDropRejected prop 1`] = `
<div>
  <Styled(t)
    accept="image/png, image/jpeg, application/pdf"
    activeStyle={
      Object {
        "borderColor": "#0099cf",
      }
    }
    className={null}
    multiple={false}
    onDrop={[Function]}
    onDropAccepted={[Function]}
    onDropRejected={[MockFunction]}
  >
    <styled.div>
      <Styled(UploadIcon) />
      <Styled(styled.p)>
        Drop here to upload or
      </Styled(styled.p)>
      <Styled(Button)
        onClick={[Function]}
        styleType="secondary"
      >
        Browse files
      </Styled(Button)>
    </styled.div>
  </Styled(t)>
</div>
`;
